### DISCUSSION TITLE
How to make aesni module default

### DISCUSSION THREAD

**Discussion by M-Stenzel** [strongswan/strongswan#2244] [Q&A]
Hi group,

I use strongswan server (latest version) on Linux x86_64.
Now I want to use the **aesni module** by default, make hardware support for AES default, resp.
I loaded the module, but in the module list there is no further documentation as how to make it default.
I guess, only LOADING the module does not necessarily mean that it gets used...

Could not find any documentation on this.

Martin.

**Comment by tobiasbrunner** [comment]
Are you referring to the kernel module (`aesni_intel`) or strongSwan's _aesni_ plugin (which would only affect IKE)? For the kernel module, you can see whether it's used e.g. in `/proc/crypto`. For strongSwan, you can see which plugin provides the implementation of an algorithm via `swanctl --list-algs` (it's determined by the plugin [load order](https://docs.strongswan.org/docs/5.9/plugins/pluginLoad.html)).

**Reply by M-Stenzel** [reply]
Hi Tobias,
well, thank you for demanding clarification.
I meant the _aesni_ plugin, not the kernel module.

I put
```
    ike=aes256gcm16-sha512-prfsha512-ecp512bp
    esp=aes256gcm16-sha512-prfsha512-ecp512bp
```
in my ipsec.conf

Plus 
_swanctl --list-algs | grep aesni_

gives me

```
  AES_CBC[aesni]
  AES_ECB[aesni]
  AES_CTR[aesni]
  AES_XCBC_96[aesni]
  AES_CMAC_96[aesni]
  AES_CCM_8[aesni]
  AES_CCM_12[aesni]
  AES_CCM_16[aesni]
  AES_GCM_8[aesni]
  AES_GCM_12[aesni]
  AES_GCM_16[aesni]
  PRF_AES128_XCBC[aesni]
  PRF_AES128_CMAC[aesni]
```

From my very limited understanding I can assume that hardware support for AES is being used?

Martin.

**Reply by tobiasbrunner** [reply]
> From my very limited understanding I can assume that hardware support for AES is being used?

Yep. But again, this only affects IKE traffic, so that won't make much of a difference.
